manifests/version.pp

```diff
@@ -11,37 +11,9 @@
   }
 
   case $::osfamily {
-    'RedHat': {
-      if ($::operatingsystem == 'Amazon') {
-        $default = '2.2'
-      } elsif ($::operatingsystem == 'Fedora' and versioncmp($distrelease, '18') >= 0) or ($::operatingsystem != 'Fedora' and versioncmp($distrelease, '7') >= 0) {
-        $default = '2.4'
-      } else {
-        $default = '2.2'
-      }
-    }
-    'Debian': {
-      if $::operatingsystem == 'Ubuntu' and versioncmp($::operatingsystemrelease, '13.10') >= 0 {
-        $default = '2.4'
-      } elsif $::operatingsystem == 'Debian' and versioncmp($distrelease, '8') >= 0 {
-        $default = '2.4'
-      } else {
-        $default = '2.2'
-      }
-    }
-    'FreeBSD': {
+    'RedHat','Debian','FreeBSD','Gentoo','Suse': {
```

Comment:
I'm not sure if we gain much from this case statement. What about removing it and hardcode `$default` to `2.4` for now? This would allow the module to maybe be compatible to other platforms.

And then, a single class that only exposes one parameter is maybe not ideal. It works fine, but feel wrong. This this be moved into params.pp / data-in-modules?

Reply:
So the main reason for this was to have the failure condition. Since we'd be dropping support from several OSes I think it's important to have that check for clarity as I expect people to be thrown by the upgrade. I should actually update the error message and make sure people know to check their OS support against the older version of the module? Otherwise I do agree with what you're saying, maybe there's a nicer way of doing it.